clients/pkg/promtail/discovery/dockerd/dockerd.go

```diff
@@ -0,0 +1,320 @@
+// Copyright 2021 The Prometheus Authors
```

Comment:
I'm sorry. I'm trying to understand how this is different from https://github.com/prometheus/prometheus/blob/main/discovery/moby/docker.go which should be used by the [current Docker target](https://github.com/grafana/loki/blob/main/clients/pkg/promtail/targets/docker/targetmanager.go#L71).

Reply:
sure ,it's able to connect promtail to the Docker daemon. the prometheus discovery
 docker_sd_configs (https://github.com/prometheus/prometheus/blob/main/discovery/moby/docker.go) couldn't list all containers of a host that just list the pod's sandbox container.

Reply:
Ah, so you are running it in Kubernetes and can only find the containers of the pod Promtail is running in?

Reply:
Another thing is docker_sd_configs of promtail, it's not work with relabel_configs.

Reply:
> Ah, so you are running it in Kubernetes and can only find the containers of the pod Promtail is running in?

No, all of the containers of the hosts that Promtail running on.it's include App container and sandbox container.

Reply:
But how is it different from the original Docker service discovery. Im asking because I'm wondering if we should rather patch the service discovery upstream.